src/Data/Tree/AVL/Proof.hs

```diff
@@ -1,24 +1,27 @@
-{-# LANGUAGE MonoLocalBinds       #-}
-{-# LANGUAGE NamedFieldPuns       #-}
-{-# LANGUAGE RankNTypes           #-}
-{-# LANGUAGE StandaloneDeriving   #-}
-{-# LANGUAGE TemplateHaskell      #-}
-{-# LANGUAGE UndecidableInstances #-}
+-- | 'Proof' datatype and its checker.
 
-module Data.Tree.AVL.Proof where
+module Data.Tree.AVL.Proof
+    ( -- * Proof
+      Proof (..)
+
+      -- * Checker
+    , checkProof
+    )
+  where
 
-import Control.Lens (makePrisms)
 import Control.Monad.Free (iter)
 
 import GHC.Generics (Generic)
 
 import Data.Tree.AVL.Internal
 
-newtype Proof h k v = Proof { unProof :: Map h k v }
-    deriving (Show,Generic)
-
-makePrisms ''Proof
+-- | Proof for operation.
```

Comment:
Better to specify that `Proof` is simply a subtree with some hashes pruned here, not in `Data.Tree.AVL.Prune` module.